cartography/models/semgrep/dependencies.py

```diff
@@ -0,0 +1,77 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from cartography.models.core.common import PropertyRef
+from cartography.models.core.nodes import CartographyNodeProperties
+from cartography.models.core.nodes import CartographyNodeSchema
+from cartography.models.core.nodes import ExtraNodeLabels
+from cartography.models.core.relationships import CartographyRelProperties
+from cartography.models.core.relationships import CartographyRelSchema
+from cartography.models.core.relationships import LinkDirection
+from cartography.models.core.relationships import make_target_node_matcher
+from cartography.models.core.relationships import OtherRelationships
+from cartography.models.core.relationships import TargetNodeMatcher
+
+
+@dataclass(frozen=True)
+class SemgrepDependencyNodeProperties(CartographyNodeProperties):
+    id: PropertyRef = PropertyRef('id')
+    lastupdated: PropertyRef = PropertyRef('lastupdated', set_in_kwargs=True)
+    name: PropertyRef = PropertyRef('name')  # TODO: set extra_index=True for this or other properties?
```

Comment:
The querybuilder will automatically create indexes if any field on a node schema is referenced by another `TargetNodeMatcher`. Only set  extra_index=True if this isn't the case _and_ you feel that it will be queried by `name` frequently in our usecases.

Reply:
AFAICT we don't have an extra index for properties besides `id` and `lastupdated` for the existing Dependency and PythonLibrary nodes, so I'll skip it here too